pywatts/modules/generation/anomaly_generation_module.py

```diff
@@ -0,0 +1,390 @@
+import numbers
+from typing import Optional, Union, Dict, List
+
+import numpy as np
+import xarray as xr
+
+from pywatts.core.base import BaseTransformer
+from pywatts.core.exceptions.wrong_parameter_exception import WrongParameterException
+
+
+class AnomalyGeneration(BaseTransformer):
+    """
+    Module for generating anomalies. Anomalies depend on their type and their length.
+    Examples are not-a-number anomalies or constant values over a certain period of time.
+    """
+
+    def __init__(self, name: str = "AnomalyGeneration", count: Union[int, float] = 1,
+                 anomaly: str = "gap", anomaly_params: Dict = {},
+                 length_params: Dict = {}, label: Optional[int] = None, seed: int = 0):
+        """
+        Initialization routine for the anomaly creation module.
+
+        :param name: Name of the module (default 'AnomalyGeneration').
+        :type name: str
+        :param count: How many anomalies should be inserted.
+                      Could be a percentage value (float) or a specific number (int) (default 1).
+        :type count: Union[int, float]
+        :param anomaly: Type of anomaly to be created. E.g. 'gap', 'outlier', 'negate' or 'constant' (default 'gap').
```

Comment:
As an alternative you could use enums instead of strings. The usage of enums has the advantage that typos are not possible, if the user specifies the anomaly.

Reply:
If I use enums, I couldn't add anomaly types in subclasses because I can simply add anomaly types by adding _anomaly_TYPE method to that class. I would you do that with enums?

Reply:
I will try it with enums.

Reply:
I updated the files to use Enums instead of strings. Enums couldn't be extended. So, EnergyAnomalyGeneration and PowerAnomalyGeneration do not have the anomaly types offered by AnomalyGeneration base class as I mentioned before.

Also, if someone renames a method, it will fail only in the transform step because the renamed method won't be found.

I had to add __init__ methods to EnergyAnomalyGeneration and PowerAnomalyGeneration because of the Types. The GAP default value for AnomalyGeneration won't be available in the EnergyAnomalyGerneration.Types. However, it is available in the subclass but not selectable due to it is not available in the enum...

Additionally, the anomaly types can have parameters that a user can add. These parameters are also wrong if the user has wrong typos in them. So, we finally did not solve it completely.

In my opinion, I would prefer the version without enums.

Reply:
Ok, you arguments convinced me. It was just an idea.